docs/source-builds.rst

```diff
@@ -0,0 +1,134 @@
+and hence beyond the scope of these build suggestions.
+
+More commonly, the components that require building are found by repeatedly
+running venvstacks and recording the list of projects that the locking failures
+indicate do not have binary wheels available for installation.
+
+Once that list of projects is available, a use case dependent mechanism can
+then be used to coordinate downloading the source artifacts and producing
+appropriately built wheels for the platforms of interest (essentially
+undertaking an ad hoc approach to the problem that Fromager approaches
+systematically).
+
+Including private artifacts in layer builds
+===========================================
+
+Once the binary wheels are available, the second task is then to actually
+include those wheels into the layer locking and building process.
+
+Private index servers
+---------------------
+
+The primary recommended approach is to run the stack builds against a private
+index server such as a self-hosted `depvi <https://doc.devpi.net/>`__ instance,
+or a cloud-hosted repository service such as
+JFrog's `Artifactory <https://jfrog.com/help/r/jfrog-artifactory-documentation/pypi-repositories>`__
+or Astral's `pyx <https://astral.sh/pyx>`__.
+
+These can be set up to serve as both a caching proxy for publicly available packages
+*and* a host for privately built packages, allowing the stack builds to be
+appropriately configured with a single ``tool.uv.index`` entry in the stack
+definition file:
+
+.. code:: toml
+
+   [[tool.uv.index]]
+   url = "https://internal.example.com/pyindex/"
+   default = true
+
+More complex arrangements using the 
+:ref:`package_indexes <package-indexes>` and
+:ref:`priority_indexes <priority-indexes>` layer specification settings are
+also possible. The
+`published examples <https://github.com/lmstudio-ai/venvstacks/tree/main/examples>`__
+demonstrate such configurations using the public PyTorch repositories,
+as those are the kinds of parallel build scenarios where the simple caching
+proxy override approach may be insufficient.
+
+.. versionadded: 0.8.0
+   Added support for ``uv`` configuration with layer specific adjustments
+   (:ref:`release details <changelog-0.8.0>`).
+
+Local wheel directories
+-----------------------
+
+Prior to the addition of index server configuration support, the only provided
+mechanism for including additional wheels in the layer locking and building
+process was to pass the ``--local-wheels`` option to the ``venvstacks`` CLI.
+
+This mechanism is still supported, with no plans to remove it, but there may
+be some situations where ``uv`` will be unable to lock a stack defined this way,
+while being able to successfully lock a stack that uses an appropriate index
+server configuration instead.
+
+.. versionchanged: 0.8.0
+   With the introduction of cross-platform layer lock files,
+   wheels for all platforms should be present on the system
```

Comment:
Wording here needs to be stronger, as layer builds will fail for any platforms that don't have wheels available at lock time.

At layer build time, the wheels also need to be available in the same location relative to the layer lock file